test/SettingsMigrator.UnitTests/MigratorUnitTests.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine.UnitTests
+{
+    using System.IO;
+    using System.Xml;
+    using Microsoft.VisualStudio.TestPlatform.CommandLine;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class MigratorUnitTests
+    {
+        [TestMethod]
+        public void MigratorGeneratesCorrectRunsettingsForEmbeddedTestSettings()
```

Comment:
Add tests for negative cases as well.
Incorrect testsettings, incorrect path etc.

Reply:
Done